src/virtual_machine/lc3_virtual_machine.rs

```diff
@@ -0,0 +1,384 @@
+use std::io::Read;
+
+use super::{instructions::*, register::Register, trap::Trap};
+
+const AMMOUNT_OF_REGISTERS: usize = 10;
+pub enum Flag {
+    POSITIVE = 1 << 0,
+    ZERO = 1 << 1,
+    NEGATIVE = 1 << 2,
+}
+
+enum MemoryMappedRegister {
+    KeyBoardStatusRegister = 0xFE00,
+    KeyBoardDataRegister = 0xFE02,
+}
+
+pub struct LC3VirtualMachine {
+    registers: Vec<u16>,
+    memory: Vec<u16>,
+}
+
+impl LC3VirtualMachine {
+    pub fn new(program_counter_start: u16) -> Self {
+        let mut registers = vec![0; AMMOUNT_OF_REGISTERS];
+        registers[Register::ProgramCounter as usize] = program_counter_start;
+        Self {
+            registers,
+            memory: vec![0; 1 << 16],
+        }
+    }
+
+    pub fn read_register(&self, source_register: Register) -> u16 {
+        self.registers[source_register as usize]
+    }
+
+    pub fn update_register(&mut self, destination_register: Register, new_register_value: u16) {
+        self.registers[destination_register as usize] = new_register_value;
+    }
```

Comment:
I'm not convinced about this. If there is a `register` module, don't its operations belong to said module? Maybe use a `Registers` struct and delegate some responsability to it.

Reply:
Yes, a `Registers` struct would make things clearer and more descriptive